lnrpc/lightning.proto

```diff
@@ -4266,11 +4296,17 @@ message ForwardingEvent {
     // second half of the circuit.
     uint64 amt_out = 6;
 
-    // The total fee (in satoshis) that this payment circuit carried.
-    uint64 fee = 7;
+    // The total fee (in satoshis) that this payment circuit carried. A negative
+    // amount is reported as zero. Deprecated, use fee_signed_msat.
+    uint64 fee = 7 [deprecated = true];
+
+    // The total fee (in milli-satoshis) that this payment circuit carried. A
+    // negative amount is reported as zero. Deprecated, use fee_signed_msat.
+    uint64 fee_msat = 8 [deprecated = true];
 
-    // The total fee (in milli-satoshis) that this payment circuit carried.
-    uint64 fee_msat = 8;
+    // The total fee (in milli-satoshis) that this payment circuit carried. This
```

Comment:
Is it possible to separate the InboundFee from the OutboundFees? As a RoutingNode operator, this information will be crucial to accurately determine the Profit & Loss for each involved channel. With the net amount, allocation will not be possible.

Reply:
Yes it is a bigger change. But also the sender doesn't indicate which part of that total fee is intended to be for the in and out sides. So it would need to be assumed based on the fee schedule at the time of forward. Then if the sender overpays, did they overpay in or out? I can see the advantage of P&L per channel though.

Reply:
>Otherwise, one would need to keep track of fee rate changes externally from which the data could be computed

Isn't this already the case currently with just outbound fees? If you change the fee rate, you need to keep track of that externally in order to possibly explain a change in fee revenue afterwards?

I don't think that splitting the fee overpayment is the best idea. It creates an impression of accuracy, where it really isn't. I'd then rather go for recording the fee schedule that was active at the time of forward. The user of this data can figure out themselves how to interpret the only value that is actually observable - the total fee. To keep the data concise, one would probably only record the changes. And with that, it is probably better to create a fee_history bucket rather than trying to embed it in the forwarding log.

Either way, it seems that this enhancement is optional and can be added later? Especially considering that it requires at least a db change too.

Reply:
> Isn't this already the case currently with just outbound fees? If you change the fee rate, you need to keep track of that externally in order to possibly explain a change in fee revenue afterwards?

To some degree, yes. For some applications it's enough to only track the volume during fixed intervals and then adjust the fee rate (which then sort of implicitly is a memory of the history of changes). Also, you can compute an estimated fee rate/base fee based on the in/out amounts, but that only works in the limits of high/low amounts (and assuming no over-payments).

> I don't think that splitting the fee overpayment is the best idea. It creates an impression of accuracy, where it really isn't. I'd then rather go for recording the fee schedule that was active at the time of forward. The user of this data can figure out themselves how to interpret the only value that is actually observable - the total fee. To keep the data concise, one would probably only record the changes. And with that, it is probably better to create a fee_history bucket rather than trying to embed it in the forwarding log.

Agree, it's then also possible to evaluate effects from base fees, a fee history bucket would be nice to have.

> Either way, it seems that this enhancement is optional and can be added later? Especially considering that it requires at least a db change too.

Sounds good!